src/bentoml/models.py

```diff
@@ -318,6 +318,37 @@ def create(
         res.exit_cloudpickle_context(imported_modules)
 
 
+@inject
+@contextmanager
+def save(
+    name: Tag | str,
+    *,
+    module: str = "",
+    api_version: str | None = None,
+    labels: dict[str, t.Any] | None = None,
+    metadata: dict[str, t.Any] | None = None,
+    _model_store: ModelStore = Provide[BentoMLContainer.model_store],
+) -> t.Generator[Model, None, None]:
+    api_version = "v1" if api_version is None else api_version
+    res = Model.create(
+        name,
+        module=module,
+        api_version=api_version,
+        labels=labels,
+        signatures={},
+        metadata=metadata,
+        context=ModelContext(framework_name="", framework_versions={}),
+    )
+
+    try:
+        yield res
```

Comment:
should we just yield the path here instead of Model?

Reply:
I think leaving it as Model is okay. Users can access values that we have auto-generated through the Model (eg. Model.tag, etc). Model also provides some useful properties (eg. Model.file_size). Though not too sure helpful they'd be to the users 😅 

imo, it may be more intuitive to have the `model_ref` rather than the `model_file_path` when running the following code:
```
with bentoml.models.save as model_ref:
    ...
# vs
with bentoml.models.save as model_file_path:
    ...
```

Reply:
@parano An example is that if user does not provide a version in the model name, the model_ref will have an auto-generated version, which may be useful for users.